### PR TITLE
WIP Correlate sent errors with received errors.

### PR DESCRIPTION
**Pull request by erights**
This is a delta to the hackathon branch as a temporary approximation of a future change.

When sending errors over marshal, generate an errorId to uniquely identify this error sending and use `assert.note` to associate that errorId with both the error as sent and the copy of that error as received. Somehow remember the association of that errorId with the sent error so that the sent error's hidden debug information --- details and stack trace --- can be correlated with the received error, should the received error be reported.

As a temporary stopgap for the hackathon, this PR takes the following shortcuts:
   * Rather than use some internal log or bookkeeping mechanism to associate the errorId with the sent error, just log the sent error to the console, so the association gets printed to the log. This is adequate but painful for human eyeballs for find the association. It is not suitable for a Causeway-like automated tool to stitch together the distributed causality graph.
   * The errorId is not very unique. We're currently only using increasing counts relative to a marshal instance.

This is currently a WIP Draft because it does not yet have new tests for this functionality.

### CODE REVIEW

**Review by michaelfig — APPROVED** [review]
LGTM, glad that tests pass. 💯

**Comment by katelynsills** [comment]
the changes look good to me, but I would need to see how they affect what the user sees.

**Review by Chris-Hibbert — APPROVED** [review]
This seems good.  How hard would it be to add a command-line flag to turn this reporting on or off?  If it's as verbose as you imply, it would be nice to only get it when someone is tracking a problem down. (Though how many of our developers would know where to turn on the flag?)

File: packages/marshal/src/marshal.js [hunk 1]
```diff
@@ -592,10 +596,21 @@ export function makeMarshal(
...
+
+              errorCount += 1;
+              // TODO in order to reliably correlate sent and received errors
+              // we will need something more globally identufying than the count
+              // relative to this marshal instance.
+              const errorId = `error#${errorCount}`;
+              assert.note(val, d`Sent as ${errorId}`);
+              // TODO we need to instead log to somewhere hidden
+              // to be revealed when correlating with the received error.
+              console.error('Temporary logging of sent error', val);
```
Comment (michaelfig) [inline packages/marshal/src/marshal.js:608]: If you make this a `console.log`, it will only display when `agoric start` is run with `-v`.

**Comment by michaelfig** [comment]
> This seems good. How hard would it be to add a command-line flag to turn this reporting on or off?

https://github.com/Agoric/agoric-sdk/pull/2037/files#r526533669

File: packages/marshal/src/marshal.js (hunk 1 above)
Comment (erights) [reply packages/marshal/src/marshal.js:608]: What does `agoric start` do or not do depending on `-v`?

What about `yarn test`? Any similarly convenient way to make it less noisy?

Comment (michaelfig) [reply packages/marshal/src/marshal.js:608]: That's what anylogger gets you (wrapping the console).  None of the tests customize it, so they are noisy.

**Comment by erights** [comment]
I report progress at https://github.com/Agoric/dapp-fungible-faucet/pull/10#issuecomment-730084320

Without the `-v` flag the experience is unchanged. With it, the experience is miserably verbose and unclear. But it does contain the needed clues, so is arguably better than the status quo.

File: packages/marshal/src/marshal.js (hunk 1 above)
Comment (erights) [reply packages/marshal/src/marshal.js:608]: Changed to `console.log`. Resolving.

**Review by katelynsills — APPROVED** [review]

**Comment by katelynsills** [comment]
Merging as per discussion in this morning's meeting.

**Pull request merged**